src/MongoDB.Driver.Core/Core/Operations/QueryHelper.cs

```diff
@@ -63,7 +63,7 @@ public static BsonDocument CreateReadPreferenceDocument(ServerType serverType, R
             if (readPreference.ReadPreferenceMode == ReadPreferenceMode.Primary || readPreference.ReadPreferenceMode == ReadPreferenceMode.SecondaryPreferred)
             {
                 var hasTagSets = readPreference.TagSets != null && readPreference.TagSets.Count > 0;
-                if (!hasTagSets && !readPreference.MaxStaleness.HasValue)
+                if (!hasTagSets && !readPreference.MaxStaleness.HasValue && readPreference.Hedge == null)
```

Comment:
[Spec](https://github.com/mongodb/specifications/blob/4e570fc442e0a67bcb4999c85a0afb10c0c5a8ae/source/server-selection/server-selection.rst#passing-read-preference-to-mongos) says:
>For mode 'secondaryPreferred', drivers MUST set the slaveOK wire protocol flag. If the read preference contains a non-empty tag_sets parameter, maxStalenessSeconds is a positive integer, or the hedge parameter is non-empty, drivers MUST use $readPreference; otherwise, drivers MUST NOT use $readPreference

As I understand it means that the provided logic should be applied only for `SecondaryPreferred`, in the current case it will be applied for `SecondaryPreferred` and `Primary` modes. Is it ok?

Reply:
As for `slaceOk` flag, based on this line https://github.com/mongodb/mongo-csharp-driver/blob/master/src/MongoDB.Driver.Core/Core/Operations/FindOpcodeOperation.cs#L467, I think slaveOk will be set correctly

Reply:
Actually looking at this spec requirement:
>For mode 'primary', drivers MUST NOT set the slaveOK wire protocol flag and MUST NOT use $readPreference

should we even be able to return anything except of null for Primary mode?